### PR TITLE
Implement a wait.wait_exponential_jitter per Google's storage guide

### PR DESCRIPTION
**Pull request by IsaacG**
Relates to #162 and #223 in that the minimum wait will never be lower than the initial/min value.

### CODE REVIEW

**Comment by IsaacG** [comment]
> warning No release notes detected. Please make sure to use [reno](https://docs.openstack.org/reno/latest/user/usage.html) to add a changelog entry.

Added. See releasenotes/notes/wait_exponential_jitter-6ffc81dddcbaa6d3.yaml

File: doc/source/index.rst [hunk 1]
```diff
@@ -205,6 +205,9 @@ exceptions, as in the cases here.
 
 .. testcode::
 
+    class ClientError(Exception):
+        """Some type of client error."""
+
```
Comment (jd) [inline doc/source/index.rst:210]: I'd put that in a different PR though.

File: tenacity/wait.py [hunk 2]
```diff
@@ -189,3 +189,37 @@ class wait_random_exponential(wait_exponential):
...
+    The wait time is min(initial * (2**n + random.uniform(0, jitter)), maximum)
+    where n is the retry count.
+    """
+
+    def __init__(
+        self,
+        initial: typing.Union[int, float] = 1,  # noqa
+        max: typing.Union[int, float] = _utils.MAX_WAIT,  # noqa
+        exp_base: typing.Union[int, float] = 2,
+        jitter: typing.Union[int, float] = 1,
```
Comment (jd) [inline tenacity/wait.py:211]: Using only `float` ought to be enough.

**Review by jd — CHANGES_REQUESTED** [review]

File: doc/source/index.rst (hunk 1 above)
Comment (IsaacG) [reply doc/source/index.rst:210]: I would, too... but checks fail without this in place. I can send this commit in first.

File: tenacity/wait.py (hunk 2 above)
Comment (IsaacG) [reply tenacity/wait.py:211]: Oh... yeah. That's a good point! Fixing.

File: doc/source/index.rst (hunk 1 above)
Comment (jd) [reply doc/source/index.rst:210]: It's ok, we'll rebase it.

File: tenacity/wait.py [hunk 3]
```diff
@@ -189,3 +189,37 @@ class wait_random_exponential(wait_exponential):
...
+    This implements the strategy described here:
+    https://cloud.google.com/storage/docs/retry-strategy
+
+    The wait time is min(initial * (2**n + random.uniform(0, jitter)), maximum)
+    where n is the retry count.
+    """
+
+    def __init__(
+        self,
+        initial: float = 1,  # noqa
```
[the remark this answers is not in the document]
Comment (IsaacG) [reply tenacity/wait.py]: Copy/pasted. Dropping.

File: tenacity/wait.py [hunk 4]
```diff
@@ -189,3 +189,37 @@ class wait_random_exponential(wait_exponential):
...
+    https://cloud.google.com/storage/docs/retry-strategy
+
+    The wait time is min(initial * (2**n + random.uniform(0, jitter)), maximum)
+    where n is the retry count.
+    """
+
+    def __init__(
+        self,
+        initial: float = 1,
+        max: float = _utils.MAX_WAIT,  # noqa
```
Comment (jd) [inline tenacity/wait.py:209]: same question here though 😁

Comment (IsaacG) [reply tenacity/wait.py:209]: I think it's the `_utils`? But I can try dropping it.

Comment (jd) [reply tenacity/wait.py:209]: I don't think flake8 cares about that 🤔

Comment (jd) [reply tenacity/wait.py:209]: AH! ok

**Review by jd — APPROVED** [review]

**Comment by IsaacG** [comment]
Thank you for the review and approval!

**Comment by IsaacG** [comment]
@jd Is there any chance I could get you to cut a new release with this new class added?
